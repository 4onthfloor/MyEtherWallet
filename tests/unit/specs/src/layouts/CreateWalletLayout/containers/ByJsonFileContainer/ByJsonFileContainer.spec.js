import Vue from 'vue';
import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import ByJsonFileContainer from '@/layouts/CreateWalletLayout/containers/ByJsonFileContainer/ByJsonFileContainer.vue';
import SuccessModal from '@/containers/ConfirmationContainer/components/SuccessModal/SuccessModal.vue';
import ByJsonBlock from '@/layouts/CreateWalletLayout/components/ByJsonBlock/ByJsonBlock.vue';
import sinon from 'sinon';
import { Tooling } from '@@/helpers';
import * as createWallet from '@/translations/create-wallet/en_US.json';

const showModal = sinon.stub();
const hideModal = sinon.stub();

const BModalStub = {
  name: 'b-modal',
  template: '<div><slot></slot></div>',
  props: ['to'],
  methods: {
    show: showModal,
    hide: hideModal
  }
};
const newObj = Object.assign({}, { createWallet: createWallet });

const translator = function(keyname) {
  return keyname.split('.').reduce((o, i) => o[i], newObj);
};

describe('ByJsonFileContainer.vue', () => {
  let localVue, i18n, wrapper, store;
  beforeAll(() => {
    const baseSetup = Tooling.createLocalVueInstance();
    localVue = baseSetup.localVue;
    i18n = baseSetup.i18n;
    store = baseSetup.store;

    Vue.config.warnHandler = () => {};
  });

  beforeEach(() => {
    wrapper = shallowMount(ByJsonFileContainer, {
      localVue,
      i18n,
      store,
      attachToDocument: true,
      stubs: {
        'router-link': RouterLinkStub,
        'by-json-block': ByJsonBlock,
        'success-modal': SuccessModal,
        'b-modal': BModalStub
      }
    });
  });

  afterAll(() => {
    wrapper.destroy();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct contents data', () => {
    const contentElements = wrapper.findAll('.contents .content-block');

    contentElements.wrappers.forEach((contentElement, i) => {
      const header = contentElement.find('h6').text();
      const pTag = contentElement.find('p').text();
      if (header && pTag) {
        expect(header).toEqual(translator(wrapper.vm.$data.contents[i].title));
        expect(pTag).toEqual(translator(wrapper.vm.$data.contents[i].desc));
      }
    });
  });

  it('should render correct downloadable data', async () => {
    wrapper.setData({ downloadable: true });
    await wrapper.vm.$nextTick();
    expect(wrapper.find('.user-input-container span').exists()).toBe(true);
    wrapper.setData({ downloadable: false });
    await wrapper.vm.$nextTick();
    expect(wrapper.find('.user-input-container span').exists()).toBe(false);
  });

  it('should render correct walletJson data', async () => {
    const walletJson = 'walletJson';
    wrapper.setData({ walletJson });
    await wrapper.vm.$nextTick();
    expect(
      wrapper.vm.$el.querySelector('.user-button a').getAttribute('href')
    ).toEqual(walletJson);
  });

  it('should render correct name data', async () => {
    const name = 'name';
    wrapper.setData({ name });
    await wrapper.vm.$nextTick();
    expect(
      wrapper.vm.$el.querySelector('.user-button a').getAttribute('download')
    ).toEqual(name);
  });

  describe('ByJsonFileContainer.vue Methods', () => {
    it('should render correct downloadDone method', () => {
      wrapper.find('.user-button a').trigger('click');
      expect(showModal.called).toBe(true);
    });
  });

  it('should dismount properly', () => {
    expect(wrapper.destroy()).toBe(undefined);
  });
});
