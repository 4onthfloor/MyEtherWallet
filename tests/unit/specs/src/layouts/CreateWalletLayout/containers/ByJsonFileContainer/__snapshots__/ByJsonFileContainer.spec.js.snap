// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ByJsonFileContainer.vue should render correctly 1`] = `
<div class="create-wallet-by-json-file">
  <div centered="" hide-footer="" hide-header="" static="" lazy="" class="bootstrap-modal no-padding">
    <div class="modal-content-block"><img src="@/assets/images/modal/garlands.png">
      <div class="d-block text-center"><i aria-hidden="true" class="check-icon fa fa-check"></i>
        <h2 class="title">
          Success
        </h2>
        <p>Your have created your wallet successfully.</p>
      </div>
      <div class="buttons">
        <standard-button-stub options="[object Object]" clickfunction="function () { [native code] }" style="display: none;"></standard-button-stub>
        <standard-button-stub options="[object Object]" clickfunction="function () { [native code] }"></standard-button-stub>
      </div>
    </div>
  </div>
  <div class="wrap">
    <div class="nav-tab-user-input-box">
      <b-tabs-stub tag="div">
        <div class="progress-bar"></div>
        <b-tab-stub active="true" tag="div" buttonid="" title="By Keystore File">
          <div class="title-block">
            <div class="title-popover">
              <h3>Save My Keystore File</h3>
            </div>
          </div>
          <div class="contents">
            <div class="content-block">
              <div class="icon-block"><img src="" alt="" class="icon"></div>
              <div class="text-block">
                <h6>Don't Lose It</h6>
                <p>Be careful, it can not be recovered if you lose it.</p>
              </div>
            </div>
            <div class="content-block">
              <div class="icon-block"><img src="" alt="" class="icon"></div>
              <div class="text-block">
                <h6>Don't Share It</h6>
                <p>Your funds will be stolen if you use this file on a malicious phishing site.</p>
              </div>
            </div>
            <div class="content-block">
              <div class="icon-block"><img src="" alt="" class="icon"></div>
              <div class="text-block">
                <h6>Make a Backup</h6>
                <p>Secure it like the millions of dollars it may one day be worth.</p>
              </div>
            </div>
          </div>
          <div class="user-input-container">
            <div class="user-input">
              <div class="user-button"><a href="" download="" rel="noopener noreferrer" class="disable next-button large-round-button-green-filled nopadding">
                  <!---->
                  <div class="generating"><i class="fa fa-spinner fa-lg fa-spin"></i>
                    <p>Please wait while we generate your keystore file...</p>
                  </div>
                </a></div>
            </div>
          </div>
        </b-tab-stub>
      </b-tabs-stub>
    </div>
  </div>
</div>
`;
